src/freeze_model.py

```diff
@@ -42,122 +46,128 @@ def strip(input_graph, drop_scope, input_before, output_after, pl_name):
+        sess.run(tf.global_variables_initializer())
+
+        saver = tf.train.Saver()
+
+                # call training script
+        with open(exp_dir / 'experiment_id_whole') as f:
+            model = f.read().rstrip()
+        results_folder = str(exp_dir / 'experiments' / model) + '/'
+        saver.restore(sess, results_folder)
+        tf_vars = [sess, normalized_y, cost, x, y_, is_train]
+
+        gd = sess.graph.as_graph_def()
+
+        for node in gd.node:
+            if node.op == 'RefSwitch':
+                node.op = 'Switch'
+                for index in range(len(node.input)):
+                    if 'moving_' in node.input[index]:
+                        node.input[index] = node.input[index] + '/read'
+            elif node.op == 'AssignSub':
+                node.op = 'Sub'
+                if 'use_locking' in node.attr: del node.attr['use_locking']
+            elif node.op == 'AssignAdd':
+                node.op = 'Add'
+                if 'use_locking' in node.attr: del node.attr['use_locking']
+            elif node.op == 'Assign':
+                node.op = 'Identity'
+                if 'use_locking' in node.attr: del node.attr['use_locking']
+                if 'validate_shape' in node.attr: del node.attr['validate_shape']
+                if len(node.input) == 2:
+                    # input0: ref: Should be from a Variable node. May be uninitialized.
+                    # input1: value: The value to be assigned to the variable.
+                    node.input[0] = node.input[1]
+                    del node.input[1]
+
+        if arch == 'vgg':
+            for i in range(1, 4):
+                drop_scope = 'model/dropout/Identity' if i == 1 else 'model/dropout_{}/Identity'.format(i)
+                input_before = 'model/max_pooling2d/MaxPool' if i == 1 else 'model/max_pooling2d_{}/MaxPool'.format(i)
+
+                gd = strip(gd, drop_scope, input_before,
+                        'model/{}CNN/Conv2D'.format(i + 1),
+                        'Placeholder_1')
+            gd = strip(gd, 'model/dropout_4/Identity', 'model/flatten/Reshape', 'model/dense/MatMul', 'Placeholder_1')
+
+        elif arch == 'musicnn':
+            for i in range(0, 2):
+                drop_scope = 'model/dropout/Identity' if i == 0 else 'model/dropout_{}/Identity'.format(i)
+
+                # the dropout layers are connected after the 9th and 19th batch normalization layers.
+                bn_i = i + 9
+                input_before = 'model/batch_normalization/batchnorm/add_1' if bn_i == 0 else 'model/batch_normalization_{}/batchnorm/add_1'.format(bn_i)
+                output_after = 'model/dense/MatMul' if i == 0 else 'model/dense_{}/MatMul'.format(i)
+                gd = strip(gd, drop_scope, input_before, output_after, 'Placeholder_1')
+
+        elif arch == 'vggish':
+            pass
+        else:
+            raise Exception('Source task not found in "{}"'.format(config['config_train']['load_model']))
```

Comment:
I think this should be a more specific exception, e.g. [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError)